CMakeLists.txt

```
@@ -0,0 +1,89 @@
+cmake_minimum_required(VERSION 3.16 FATAL_ERROR)
+
+option(BUILD_TESTS "Build tests" OFF)
+option(BUILD_BENCHMARKS "Build benchmarks" OFF)
+option(USE_VCPKG "Use vcpkg for dependencies" OFF)
+
+if(COMMAND cmake_policy)
+    cmake_policy(SET CMP0003 NEW)
+endif(COMMAND cmake_policy)
+
+if (USE_VCPKG)
+    set(CMAKE_TOOLCHAIN_FILE ${CMAKE_CURRENT_SOURCE_DIR}/vcpkg/scripts/buildsystems/vcpkg.cmake
+    CACHE STRING "Vcpkg toolchain file")
```

[detail] I would indent here.

```suggestion
        CACHE STRING "Vcpkg toolchain file")
```